src/ExadelMentorship.Jobs/ReportJob.cs

```diff
@@ -42,14 +45,29 @@ protected async override Task ExecuteAsync(CancellationToken stoppingToken)
         //System.NotSupportedException: 'Only public methods can be invoked in the background'
         public async Task Execute(ReportUser reportUser)
         {
+            var client = new HttpClient
+            {
+            };
+
+            var disco = await client.GetDiscoveryDocumentAsync("https://localhost:7046");
+            var tokenResponse = await client.RequestClientCredentialsTokenAsync(new ClientCredentialsTokenRequest
+            {
+                Address = disco.TokenEndpoint,
+                ClientId = "reportJob",
+                ClientSecret = "secret",
+                Scope = "mailApi"
+            });
+
+            var apiClient = _httpClientFactory.CreateClient();
+            apiClient.SetBearerToken(tokenResponse.AccessToken);
+
             var report = await _reportServices.GenerateReportForUser(reportUser.Id);
             Message message = new Message(reportUser.Email, reportUser.UserName, report);
+            HttpContent httpContent = new StringContent(JsonConvert.SerializeObject(message), Encoding.UTF8, Application.Json);
```

Comment:
Is there a way, i.e. some built-in extension method somewhere that sends C# object in HTTP Post/Put request directly?

I mean it is a common use-case. There must be some convenience way already built-in.

Reply:
have not found in google yet

Reply:
Try `System.Net.Http.Json`

Reply:
Thanks, done.